src/core/thread/network_data_local.cpp

```diff
@@ -69,6 +90,30 @@ Error Local::AddOnMeshPrefix(const OnMeshPrefixConfig &aConfig)
     // of an IEEE 802.15.4 interface MUST have a length of 64 bits.
     VerifyOrExit(!aConfig.mSlaac || aConfig.mPrefix.mLength == OT_IP6_PREFIX_BITSIZE, error = kErrorInvalidArgs);
 
+    SuccessOrExit(error = ValidatePrefix(aConfig.GetPrefix()));
+    SuccessOrExit(error = ValidatePreference(aConfig.mPreference));
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    if (aConfig.mDp)
+    {
+        // Check if domain prefix is already configured
+        OnMeshPrefixConfig prefixConfig;
+        ExitNow(error = ((Get<BackboneRouter::Local>().GetDomainPrefix(prefixConfig) == kErrorNotFound) ? kErrorNone : kErrorAlready);
+    }
+#endif
```

Comment:
I think it's fine to add a domain prefix when there is already one configured (same as before).

Reply:
@simonlingoogle , @librasungirl, @wgtdkp @abtink  I have removed checking if domian prefix is already configured. The PR has been rebased. I think the wpantund should also allow to add domain prefix if the one is already configured. Currently the old one must be removed to add a new one.